src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSource.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved. 
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.Sarif.Viewer
+{
+    internal class FixSuggestedActionsSource : ISuggestedActionsSource
+    {
+        public FixSuggestedActionsSource(FixSuggestedActionsSourceProvider fixSuggestedActionsSourceProvider, ITextView textView, ITextBuffer textBuffer)
```

Comment:
>FixSuggestedActionsSourceProvider fixSuggestedActionsSourceProvider, [](start = 41, length = 68)

I would personally avoid circular references and just pass this class only what it needs.

Reply:
I'm sympathetic to that argument, but the fact that it's `public` in the sample I copied suggests that somebody in VS needs it. I'd rather leave the argument and the public property in place for now, the remove them and wonder why it mysteriously fails. I'll leave a comment to remove it once we actually see the light bulbs.

---
In reply to: [509479052](https://github.com/microsoft/sarif-visualstudio-extension/pull/228#discussion_r509479052) [](ancestors = 509479052)